client/my-sites/email/email-providers-stacked-comparison/provider-cards/utils.tsx

```diff
@@ -12,6 +12,18 @@ export enum IntervalLength {
 	MONTHLY = 'monthly',
 }
 
+export const castIntervalLength = ( stringValue: string ): IntervalLength => {
+	const isStringValueInEnum = Object.values( IntervalLength ).some(
+		( enumValue: string ) => enumValue === stringValue
+	);
+
+	if ( isStringValueInEnum ) {
+		return stringValue as IntervalLength;
+	}
+
+	return IntervalLength.ANNUALLY;
+};
```

Comment:
At a high level, I would prefer this cast function to return a valid interval or null. That way the opinion of this code doesn't bleed out into other logic, which may want to default a different value. If we update the relevant components to accept `IntervalLength | null`, we can have the component provide the default value inline.
```suggestion
export const castIntervalLength = ( stringValue: string ): IntervalLength| null => {
	const isStringValueInEnum = Object.values( IntervalLength ).some(
		( enumValue: string ) => enumValue === stringValue
	);

	if ( isStringValueInEnum ) {
		return stringValue as IntervalLength;
	}

	return null;
};
```

Reply:
You make a very good point, I'll change that 👍.